src/api/python/_api.py

```diff
@@ -165,32 +237,60 @@ def register_memory(
         reg_descs = self.get_reg_descs(reg_list, mem_type, is_sorted)
 
-        # based on backend type and mem_type, figure what registrations are meaningful
         handle_list = []
         for backend_string in backends:
             handle_list.append(self.backends[backend_string])
         self.agent.registerMem(reg_descs, handle_list)
 
         return reg_descs
 
-    # The output from get_reg_descs (which is later passed to register_memory for
-    # registration) or direct output of register_memory is passed here
+    """
+    @brief Deregister memory regions from the specified backends.
+
+    @param dereg_list nixlRegDList of memory to deregister, from register_memory or get_reg_descs.
+    @param backends Optional list of backend names for deregistration, otherwise NIXL will deregister
+            with all the backends that have these memory regions registered.
+    """
+
     def deregister_memory(
         self, dereg_list: nixlBind.nixlRegDList, backends: list[str] = []
     ):
-        # based on backend type and mem_type, figure what deregistrations are needed
         handle_list = []
         for backend_string in backends:
             handle_list.append(self.backends[backend_string])
         self.agent.deregisterMem(dereg_list, handle_list)
 
-    # Optional proactive make connection
+    """
+    @brief Optional, to proactively establish a connection with a remote agent,
+            which will reduce the time spent in the first transfer between the two agents.
+            NIXL will establish the connection for all the backends that talk to remote agents.
+
+    @param remote_agent Name of the remote agent.
+    """
+
     def make_connection(self, remote_agent: str):
         self.agent.makeConnection(remote_agent)
 
-    # agent_name should be "NIXL_INIT_AGENT" for local descriptors on the initiator side. For a
-    # remote agent, it would be that agent's name, or for loopback, local agent's name as target.
-    # xfer_list can be any of the types supported by get_xfer_descs
+    """
+    @brief Prepare a transfer descriptor list for data transfer.
+            Later elements from this list can be used to create a transfer request by index.
+            It should be done on the initiator agent, and for both sides of an transfer.
+            Considering loopback, there are 3 modes for agent_name:
+              - For local descriptors, it is set to NIXL_INIT_AGENT,
+                indicating that this is a local preparation to be used as local_side handle.
+              - For remote descriptors: it is set to the remote name, indicating
+                that this is remote side preparation to be used for remote_side handle.
+              - For loopback descriptors, it is set to local agent's name, indicating that
+                this is for a loopback (local) transfer to be uued for remote_side handle
+
+    @param agent_name Name of the agent, can be "NIXL_INIT_AGENT", local agent name, or remote agent name
```

Comment:
Not sure what's the convention but I think having a period reads better: `agent. It can`